packages/contracts/test/exchange/internal.ts

```diff
@@ -0,0 +1,329 @@
+import { BlockchainLifecycle } from '@0xproject/dev-utils';
+import { Order, SignedOrder } from '@0xproject/types';
+import { BigNumber } from '@0xproject/utils';
+import * as _ from 'lodash';
+
+import { TestExchangeInternalsContract } from '../../generated_contract_wrappers/test_exchange_internals';
+import { artifacts } from '../utils/artifacts';
+import { getGanacheOrGethError } from '../utils/assertions';
+import { chaiSetup } from '../utils/chai_setup';
+import { bytes32Values, testCombinatoriallyWithReferenceFuncAsync, uint256Values } from '../utils/combinatorial_utils';
+import { constants } from '../utils/constants';
+import { FillResults } from '../utils/types';
+import { provider, txDefaults, web3Wrapper } from '../utils/web3_wrapper';
+
+chaiSetup.configure();
+const blockchainLifecycle = new BlockchainLifecycle(web3Wrapper);
+
+const MAX_UINT256 = new BigNumber(2).pow(256).minus(1);
+const MAX_BYTES32_STRING_LENGTH = 64 + 2; // '0x' prefix + 64 hex characters
+
+const emptyOrder: Order = {
+    senderAddress: constants.NULL_ADDRESS,
+    makerAddress: constants.NULL_ADDRESS,
+    takerAddress: constants.NULL_ADDRESS,
+    makerFee: new BigNumber(0),
+    takerFee: new BigNumber(0),
+    makerAssetAmount: new BigNumber(0),
+    takerAssetAmount: new BigNumber(0),
+    makerAssetData: '0x',
+    takerAssetData: '0x',
+    salt: new BigNumber(0),
+    exchangeAddress: constants.NULL_ADDRESS,
+    feeRecipientAddress: constants.NULL_ADDRESS,
+    expirationTimeSeconds: new BigNumber(0),
+};
+
+const emptySignedOrder: SignedOrder = {
+    ...emptyOrder,
+    signature: '',
+};
+
+const overflowErrorForCall = new Error('UINT256_OVERFLOW');
+
+async function _getInvalidOpcodeErrorForCall(): Promise<Error> {
+    const errMsg = await getGanacheOrGethError('invalid opcode', 'Contract call failed');
+    return new Error(errMsg);
+}
+
+async function _getOverflowErrorForSendTransaction(): Promise<Error> {
+    const errMsg = await getGanacheOrGethError('UINT256_OVERFLOW', 'always failing transaction');
+    return new Error(errMsg);
+}
```

Comment:
I think these should be part of the utils. We don't want to change every test file when we add Parity or some other client.

But synchronizing the error objects between the different clients is a challenge. Fortunately there are only a couple of errors we should be concerned with (OutOfGas, InvalidOpcode, InvalidJump, Revert(message), etc.)